tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc

```diff
@@ -400,6 +406,10 @@ void TrtShapeOptimizationProfile::SetShapeTensorMask(
 // definition or the engine would give concrete answers.
 void TrtShapeOptimizationProfile::SetShapeTensorMask(
     const std::vector<PartialTensorShape>& input_partial_shapes) {
+  if (!is_shape_tensor_.empty()) {
+    // Already initialized, e.g. by TRTEngineOp::ComputeAsync().
```

Comment:
does it make sense to check is_shape_tensor_.size() == input_partial_shapes.size() here?